tests/test-common.js

```diff
@@ -4,949 +4,787 @@
-      });
 
-      context('AuthenticationProofPurpose', () => {
-        it('should detect an expired date', async () => {
+        it('should sign a document w/o security context', async () => {
           const Suite = suites[suiteName];
-          const signSuite = new Suite({
-            ...mock.suites[suiteName].parameters.sign,
-            date: new Date('01-01-1970')
-          });
-          const testDoc = clone(mock.securityContextTestDoc);
+          const suite = new Suite(mock.suites[suiteName].parameters.sign);
+          const testDoc = clone(mock.nonSecurityContextTestDoc);
           const signed = await jsigs.sign(testDoc, {
             documentLoader: testLoader,
-            suite: signSuite,
-            purpose: new AuthenticationProofPurpose({
-              challenge: 'abc',
-              domain: 'example.com'
-            })
+            suite,
+            purpose: suite.legacy ?
+              new PublicKeyProofPurpose() : new NoOpProofPurpose()
           });
+          let expected = mock.suites[suiteName].nonSecurityContextSigned;
+          if(pseudorandom.includes(suiteName)) {
+            expected = clone(expected);
+            if(suite.legacy) {
+              expected[constants.SECURITY_SIGNATURE_URL]
+                ['https://w3id.org/security#signatureValue'] =
+                signed[constants.SECURITY_SIGNATURE_URL]
+                  ['https://w3id.org/security#signatureValue'];
+            } else {
+              expected[constants.SECURITY_PROOF_URL]['@graph']
+                ['https://w3id.org/security#jws'] =
+                signed[constants.SECURITY_PROOF_URL]['@graph']
+                  ['https://w3id.org/security#jws'];
+            }
+          }
+          assert.deepEqual(signed, expected);
+        });
 
-          const verifySuite = new Suite(
-            mock.suites[suiteName].parameters.verify);
+        it('should verify a document w/security context', async () => {
+          const Suite = suites[suiteName];
+          const suite = new Suite(mock.suites[suiteName].parameters.verify);
+          const signed = mock.suites[suiteName].securityContextSigned;
           const result = await jsigs.verify(signed, {
             documentLoader: testLoader,
-            suite: verifySuite,
-            purpose: new AuthenticationProofPurpose({
-              challenge: 'abc',
-              domain: 'example.com',
-              date: new Date('01-01-2018'),
-              maxTimestampDelta: 0
-            })
+            suite,
+            purpose: suite.legacy ?
+              new PublicKeyProofPurpose() : new NoOpProofPurpose()
           });
+          const property = suite.legacy ? 'signature' : 'proof';
+          const expectedPurposeResult = {
+            Ed25519Signature2018: {
+              valid: true
+            },
+            RsaSignature2018: {
+              valid: true
+            },
+            EcdsaKoblitzSignature2016: {
+              controller: {
```

Comment:
This `proofResult` section for the legacy suites doesn't quite look right. As in, why is the key type RsaVerificationKey2018, on line 384, etc.